src/app/productConfig/productConfigModal/productConfig.modal.component.js

```diff
@@ -35,69 +20,57 @@ export const giveGiftParams = {
 class ProductConfigModalController {
 
   /* @ngInject */
-  constructor( $window, $location, $scope, $log, designationsService, cartService, modalStateService, analyticsFactory ) {
+  constructor( $window, $location, modalStateService ) {
     this.$window = $window;
     this.$location = $location;
-    this.$scope = $scope;
-    this.$log = $log;
-    this.designationsService = designationsService;
-    this.cartService = cartService;
     this.modalStateService = modalStateService;
-    this.possibleTransactionDays = possibleTransactionDays;
-    this.startDate = startDate;
-    this.analyticsFactory = analyticsFactory;
-
-    this.selectableAmounts = [50, 100, 250, 500, 1000, 5000];
   }
 
   $onInit(){
     this.initModalData();
     this.initializeParams();
-    this.setDefaultAmount();
-    this.waitForFormInitialization();
+  }
+
+  $onDestroy(){
+    // Clear the modal name and params when the modal closes
+    this.modalStateService.name( null );
+    angular.forEach( giveGiftParams, ( value ) => {
+      // Remove all query params except CampaignCode
+      if( value !== giveGiftParams.campaignCode ) {
+        this.$location.search(value, null);
+      }
+    } );
   }
 
   initModalData() {
-    this.productData = this.resolve.productData;
+    this.code = this.resolve.code;
     this.itemConfig = this.resolve.itemConfig;
     this.isEdit = this.resolve.isEdit;
     this.uri =  this.resolve.uri;
-    this.suggestedAmounts =  this.resolve.suggestedAmounts;
-    this.nextDrawDate = this.resolve.nextDrawDate;
-
-    if(!this.itemConfig['recurring-day-of-month'] && this.nextDrawDate) {
-      this.itemConfig['recurring-day-of-month'] = startDate(null, this.nextDrawDate).format('DD');
-    }
-
-    this.showRecipientComments = !!this.itemConfig['recipient-comments'];
-    this.showDSComments = !!this.itemConfig['donation-services-comments'];
-
-    this.useSuggestedAmounts = !isEmpty(this.suggestedAmounts);
   }
 
   initializeParams() {
     if(this.isEdit){
       return;
     }
 
-    let params = this.$location.search();
-
+    // Add query params for current modal
     this.modalStateService.name( 'give-gift' );
-    this.$location.search( giveGiftParams.designation, this.productData.code );
-    this.$location.search( giveGiftParams.campaignPage, this.itemConfig['campaign-page'] );
+    this.updateQueryParam( giveGiftParams.designation, this.code ); // TODO: can this.code change? Do we need to set this again if productData comes back with a different code?
```

Comment:
This used to rely on a request for `productData`. Is there any reason the productData request would return a different code than the one it was called with? Is this how redirecting designations works? Do we care if the query param stays out of date? Even if it is redirecting, it seems like if the old query param was kept, it would just perform the redirect again to get new data on every page load.